### PR TITLE
feat(event-targets): add support for fargate/awsvpc tasks

### PR DESCRIPTION
**Pull request by jogold**
The target is "enhanced" using an AwsCustomResource calling `CloudWatchEvents.putTargets`.

Fix wrong reference to container name in `containerOverrides` (must be `name`).

BREAKING CHANGE: `targets.EcsEc2Task` renamed to `targets.EcsTask`

----

### Pull Request Checklist

* [x] Testing
  - Unit test added (prefer not to modify an existing test, otherwise, it's probably a breaking change)
  - __CLI change?:__ coordinate update of integration tests with team
  - __cdk-init template change?:__ coordinated update of integration tests with team
* [ ] Docs
  - __jsdocs__: All public APIs documented
  - __README__: README and/or documentation topic updated
  - __Design__: For significant features, design document added to `design` folder
* [x] Title and Description
  - __Change type__: title prefixed with **fix**, **feat** and module name in parens, which will appear in changelog
  - __Title__: use lower-case and doesn't end with a period
  - __Breaking?__: last paragraph: "BREAKING CHANGE: <describe what changed + link for details>"
  - __Issues__: Indicate issues fixed via: "**Fixes #xxx**" or "**Closes #xxx**"
* [ ] Sensitive Modules (requires 2 PR approvers)
  - IAM Policy Document (in @aws-cdk/aws-iam)
  - EC2 Security Groups and ACLs (in @aws-cdk/aws-ec2)
  - Grant APIs (only if not based on official documentation with a reference)

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

File: packages/@aws-cdk/aws-events-targets/lib/ecs-task-properties.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ export interface ContainerOverride {
   /**
    * Name of the container inside the task definition
    */
-  readonly containerName: string;
+  readonly name: string;
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-events-targets/lib/ecs-task-properties.ts]: I would prefer for the parameter to be called `containerName`. My apologies if it renders incorrectly, but that could and should be a different concern.

File: packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    const taskDefinitionArn = this.taskDefinition.taskDefinitionArn;
+
+    // Use a custom resource to "enhance" the target with network configuration
+    // when using awsvpc network mode.
+    if (this.taskDefinition.networkMode === ecs.NetworkMode.AwsVpc) {
+      const subnetSelection = this.props.subnetSelection || { subnetType: ec2.SubnetType.Private };
+      const assignPublicIp = subnetSelection.subnetType === ec2.SubnetType.Private ? 'DISABLED' : 'ENABLED';
+
+      new cloudformation.AwsCustomResource(this.taskDefinition, 'PutTargets', {
+        onUpdate: { // We don't need an onDelete here because the target will be owned by CF anyway
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts]: This is confusing to me. Why no `onCreate`?

Comment (jogold) [reply packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts]: `onCreate` defaults to `onUpdate`

Comment (jogold) [reply packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts]: https://github.com/awslabs/aws-cdk/blob/master/packages/%40aws-cdk/aws-cloudformation/lib/aws-custom-resource.ts#L83

File: packages/@aws-cdk/aws-events-targets/lib/ecs-task-properties.ts (hunk 1 above)
Comment (jogold) [reply packages/@aws-cdk/aws-events-targets/lib/ecs-task-properties.ts]: Ok, will update

File: packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts (hunk 2 above)
Comment (rix0rrr) [reply packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts]: Got it.

Comment (rix0rrr) [reply packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts]: I have to say I have mixed feelings about this implementation... but I guess this is the point of the CDK :)

Comment (jogold) [reply packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts]: Which implementation, `AwsCustomResource` or this ECS target?

Comment (rix0rrr) [reply packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts]: Yes, the fact that it is using a custom resource to achieve its goal.

**Pull request merged**